### PR TITLE
sync search query with `search` url param

### LINKED ISSUE #17: Add search query to address bar
in `js/archive.js`

on submit of search form, add search to the browser's location
clear it on empty search
on page load, put search query from location into the `search` variable

### PR DESCRIPTION
**Pull request by wkentdag**
Adds a `?search=xxx` query parameter that's bound to the search input. So we can link to search terms.

Closes #17

### CODE REVIEW

File: js/archive.js [hunk 1]
```diff
@@ -11,6 +11,15 @@
 
   // this runs after the DOM has loaded
   $(function() {
+    try {
+      if (!window.location.search) return;
```
Comment (tim-phillips) [inline js/archive.js]: Should probably remove this line, it stops function execution...

File: js/archive.js [hunk 2]
```diff
@@ -11,6 +11,15 @@
...
+      const { search: params } = window.location
```
Comment (tim-phillips) [inline js/archive.js]: Destructuring assignment won't work in IE 

https://caniuse.com/#feat=mdn-javascript_operators_destructuring

File: js/archive.js [hunk 3]
```diff
@@ -110,4 +123,13 @@
...
   }
+
+  function slugify(text) {
+    return text.toString().toLowerCase()
+      .replace(/\s+/g, '-')           // Replace spaces with -
+      .replace(/[^\w\-]+/g, '')       // Remove all non-word chars
+      .replace(/\-\-+/g, '-')         // Replace multiple - with single -
+      .replace(/^-+/, '')             // Trim - from start of text
+      .replace(/-+$/, '');            // Trim - from end of text
+  }
```
Comment (tim-phillips) [inline js/archive.js:148]: Slugifying will remove information that we can't get back. We might want to use [`encodeURIComponent()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent) and [`decodeURIComponent()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/decodeURIComponent) instead. What do you think?

**Comment by tim-phillips** [comment]
Thanks for starting this! It'll open up lots of possibilities 🔮

File: js/archive.js [hunk 4]
```diff
@@ -44,7 +53,7 @@
     listing.innerHTML = ''
   }
 
-  function getShows(page, search) {
+  function getShows(page, search = '') {
```
Comment (tim-phillips) [inline js/archive.js]: https://caniuse.com/#feat=mdn-javascript_functions_default_parameters

it's gonna be nice when we give ourselves a build step

File: js/archive.js (hunk 3 above)
Comment (wkentdag) [reply js/archive.js:148]: I like having clean urls but I guess it doesn't make a ton of sense in this case. Could be cool in the future though to have something like `/shows/:show-id`

Comment (tim-phillips) [reply js/archive.js:148]: That seems like a worthwhile consideration, I like the idea of nice looking show urls. 

What if we did `text.toString().trim().toLowerCase().replace(/\s/g, '-')` (i removed `+`) and then encode? If the user doesn’t want non word chars or two hyphens then they could just edit the input.

Comment (wkentdag) [reply js/archive.js:148]: updated!

Comment (tim-phillips) [reply js/archive.js:148]: This works beautifully.

**Review by tim-phillips — APPROVED** [review]

**Comment by tim-phillips** [comment]
Thank you!

**Pull request merged**